Cython/Compiler/ExprNodes.py

```diff
@@ -2736,7 +2742,7 @@ def analyse_types(self, env):
             ]))
 
     def type_dependencies(self, env):
-        return self.sequence.type_dependencies(env)
+        return self.sequence.type_dependencies(self.outer_scope if self.outer_scope else env)
```

Comment:
You're using `or` everywhere else.
```suggestion
        return self.sequence.type_dependencies(self.outer_scope or env)
```